zerver/actions/message_delete.py

```diff
@@ -103,3 +107,35 @@ def do_delete_messages_by_sender(user: UserProfile) -> None:
     )
     if message_ids:
         move_messages_to_archive(message_ids, chunk_size=retention.STREAM_MESSAGE_BATCH_SIZE)
+
+
+def classify_messages(messages: Iterable[Message]) -> tuple[list[Message], list[Message]]:
+    private_stream_messages = []
+    public_stream_messages = []
+
+    for message in messages:
+        if message.recipient.type == Recipient.STREAM:
+            stream = Stream.objects.get(id=message.recipient.type_id)
```

Comment:
I am not sure if I mentioned this before, but this will result in DB queries in a loop which is not efficient.

Reply:
We do not need whole stream objects, so would be good if we could just fetch fields needed like or maybe we can just fetch IDs of all public streams and then check whether the stream ID is in that set or not.

Reply:
And there is still a query in a loop for fetching `Stream` object which will be not needed after the above change.